vignettes/tcplfit2-vignette.Rmd

```diff
@@ -161,3 +166,159 @@ The input data for this example contains 6 signatures for one chemical in a tran
+
+* m0id = Level 0 id
+* spid = Sample id
+* acid = Unique assay component id; unique numeric id for each assay component
+* apid = Assay plate id
+* coli = Column index (location on assay plate)
+* rowi = Row index (location on assay plate)
+* wllt = well type
+* wllq = well quality
+* conc = concentration
+* rval = raw value
+* srcf = Source file name
+* clowder_uid = clowder unique id for source files
+* git_hash = hash key for pre-processing scripts
+
+```{r example4_init, fig.height = 6, fig.width = 7, message=FALSE, warning = FALSE, echo = FALSE}
+load(file = "../data/mc0.rda")
+library(tcpl)
+library(data.table)
+dat <- mc0
+DT::datatable(head(dat[wllt=='t',]),rownames= FALSE, options = list(scrollX = T))
+```
+To run standalone `tcplfit2` fitting without the need for a MySQL database connection like `invitrodb`, the user will replicate stepping through the multiple levels of processing. A detailed explanation of processing levels can be found within `tcpl`'s Data Processing vignette.
+
+Level 1 importantly establishes the concentration index. The concentration index is simply the distinct concentrations ranked from lowest to highest, and this index can be used to calculate the baseline median absolute deviation for an assay.
+
+```{r example4_cndx, fig.height = 6, fig.width = 7}
+## Order by the following columns
+setkeyv(dat, c('acid', 'srcf', 'apid', 'coli', 'rowi', 'spid', 'conc'))
+
+## Define replicate id (rpid) column for test compound wells
+nconc <- dat[wllt == "t" , ## denotes test well as the well type (wllt)
+             list(n = lu(conc)),
+             by = list(acid, apid, spid)][ , list(nconc = min(n)), by = acid]
+dat[wllt == "t" & acid %in% nconc[nconc > 1, acid],
+    rpid := paste(acid, spid, wllt, srcf, apid, "rep1", conc, sep = "_")]
+dat[wllt == "t" & acid %in% nconc[nconc == 1, acid],
+    rpid := paste(acid, spid, wllt, srcf, "rep1", conc, sep = "_")]
+
+## Define rpid column for non-test compound wells
+dat[wllt != "t",
+    rpid := paste(acid, spid, wllt, srcf, apid, "rep1", conc, sep = "_")]
+
+## set repid based on rowid
+dat[, dat_rpid := rowid(rpid)]
+dat[, rpid := sub("_rep[0-9]+.*", "",rpid, useBytes = TRUE)]
+dat[, rpid := paste0(rpid,"_rep",dat_rpid)]
+
+# Define concentration index
+indexfunc <- function(x) as.integer(rank(unique(x))[match(x, unique(x))])
+dat[ , cndx := indexfunc(conc), by = list(rpid)]
+```
+
+Levels 2 and 3 are used for data adjustments and normalization. 
+
+```{r example4_mc2, fig.height = 6, fig.width = 7}
+# If no adjustments are required for the data, the corrected value (cval) should be set as original rval
```

Comment:
I would just add a "general step section" header here. "## Adjustments" Or something to that effect.

Reply:
added

Reply:
This looks good.